commit 05e6eaa27ec6d5b598ee48c82a7a177596b39927 (refs/changes/84/5084/6)
Author: Tim Foster <tim.foster@joyent.com>
Date:   2018-11-21T18:26:28+00:00 (11 months ago)
    
    MANTA-4024 madtom doesn't use its nodeunit dependency
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/Makefile b/Makefile
index 27176bc..9be8549 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2017, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 #
@@ -25,7 +25,6 @@
 #
 # Tools
 #
-NODEUNIT        := ./node_modules/.bin/nodeunit
 NPM             := npm
 
 #
@@ -74,17 +73,15 @@ NPM_ENV		 = MAKE_OVERRIDES="CTFCONVERT=/bin/true CTFMERGE=/bin/true"
 # Repo-specific targets
 #
 .PHONY: all
-all: $(SMF_MANIFESTS) | $(NODEUNIT) $(REPO_DEPS) scripts
-	$(NPM) install
-$(NODEUNIT): | $(NPM_EXEC)
+all: $(SMF_MANIFESTS) | $(REPO_DEPS) scripts
 	$(NPM) install
 
 CLEAN_FILES += $(NODEUNIT) ./node_modules/nodeunit
 
 .PHONY: test
-test: $(NODEUNIT)
+test:
 	mkdir -p ./tmp
-	find test/ -name '*.test.js' | xargs -t -L1 ctrun -o noorphan node
+	find test/ -name '*.test.js' | xargs -t -L1 ctrun -o noorphan $(NODE_EXEC)
 
 .PHONY: scripts
 scripts: deps/manta-scripts/.git
diff --git a/package.json b/package.json
index 3992d81..93a91bf 100644
--- a/package.json
+++ b/package.json
@@ -16,8 +16,9 @@
                 "sdc-clients": "~9.1",
                 "vasync": "1.5.0"
         },
+        "//": "One of our modules has a '*' dependency on 'debug' that uses syntax incompatible with our prebuilt node version. Force it to an older version.",
         "devDependencies": {
-                "nodeunit": "0.7.4"
+                "debug": "3.1.0"
         },
         "engines": {
                 "node": "0.8.18"
